src/js/_enqueues/lib/nav-menu.js

```diff
@@ -876,6 +876,35 @@
 					$( '#submit-customlinkdiv' ).trigger( 'click' );
 				}
 			});
+
+			$('#submit-customlinkdiv').on('click', function(e) {
+				var urlInput = $('#custom-menu-item-url'),
+					url = urlInput.val(),
+					errorMessage = $('#custom-url-error'),
+					urlWrap = $('#menu-item-url-wrap');
+			
+				// Hide the error message initially
+				errorMessage.hide();
+				urlWrap.removeClass('has-error');
+			
+				if ('' === url || 'https://' === url || 'http://' === url) {
+					e.preventDefault();
+					urlInput.addClass('form-invalid')
+						.attr('aria-invalid', 'true')
+						.attr('aria-describedby', 'custom-url-error');
+			
+					// Show the error message and add the class for extra margin
+					errorMessage.show();
+					urlWrap.addClass('has-error');
+			
+					// Announce error message via screen reader
+					wp.a11y.speak('Invalid URL. Please enter a valid URL.', 'assertive');
+				} else {
+					urlInput.removeClass('form-invalid')
```

Comment:
While it makes sense for the error message to fire on submission (since the validation cases pretty much only cover the case where the input field was ignored), the error clearing state should fire on `keyup`, so the error is cleared as soon as any value that isn't `http://` or `https://` is entered.